src/components/Firebase/firebase.js

```diff
@@ -93,6 +94,32 @@ class Firebase {
       url: config.confirmationEmailRedirect
     });
 
+  // *** Algolia API ***
+  onBooksAddedListener = () =>
+    this.db.collection('books').onSnapshot(snap => {
+      snap.docChanges().forEach(change => {
+        if (change.type === 'added') {
+          console.log('change.doc', change.doc);
+          this.addOrUpdateIndexRecord(change.doc);
+        }
+      });
+    });
+
+  // TODO This is not so efficient: for some reason Firestore thinks
+  // that all its book objects have changed and then sends all of them
+  // to Algolia
+  addOrUpdateIndexRecord(dataSnapshot) {
+    let firebaseObject = dataSnapshot.data();
+    // Specify Algolia's objectID using the Firebase object key
+    firebaseObject.objectID = dataSnapshot.id;
+    // Add or update object
+    index.saveObject(firebaseObject, err => {
+      if (err) {
```

Comment:
If these codes are meant to make a thumbnail for a book. Then just add them when uploading a book, instead of here 
 where we try to get all data.

Reply:
That is exactly what is happening. Here I define them as methods of the Firebase class, such that they are available to any component that is enriched by the withFirebase HOC. It is added here for re-usability purposes of the code.